providers/collection_package.rb

```diff
@@ -8,13 +8,40 @@ def whyrun_supported?
 action :create do
   if @current_resource.exists
     Chef::Log.info "#{@new_resource} already exists - nothing to do."
+		if @current_resource.different
+			  converge_by("Update #{@new_resource}") do
+				  update_collection_package
+			  end
+		else
+			Chef::Log.info "#{@new_resource} hasn't changed - nothing to do."
+		end
   else
     converge_by("Create #{@new_resource}") do
       create_collection_package
     end
   end
 end
 
+action :create_if_missing do
+	if @current_resource.exists
+		Chef::Log.info "#{@new_resource} already exists - nothing to do."
+	else
+		converge_by("Create #{@new_resource}") do
+			create_collection_package
+		end
+	end
+end
+
+action :delete do
+	if !@current_resource.exists
+		Chef::Log.info "#{@new_resource} doesn't exist - nothing to do."
+	else
+		converge_by("Delete #{@new_resource}") do
+			delete_collection_package
+		end
+	end
+end
+
 def load_current_resource
   @current_resource = Chef::Resource.resource_for_node(:opennms_collection_package, node).new(@new_resource.name)
 
```

Comment:
You aren't setting the :different variable here, which you need to do otherwise this will never actually update anything. To do so you'll need to set all the properties of `@current_resource` to the values that `@new_resource` has, then perform comparisons to how the resource currently exists on the system. Here's an example, from the event resource:
```
def load_current_resource
  @current_resource = Chef::Resource.resource_for_node(:opennms_event, node).new(@new_resource.name)
  @current_resource.uei(@new_resource.uei || @new_resource.name)
  @current_resource.file(@new_resource.file)
  @current_resource.mask(@new_resource.mask)
  @current_resource.event_label(@new_resource.event_label)
  @current_resource.descr(@new_resource.descr)
  @current_resource.logmsg(@new_resource.logmsg)
  @current_resource.logmsg_dest(@new_resource.logmsg_dest)
  @current_resource.logmsg_notify(@new_resource.logmsg_notify)
  @current_resource.severity(@new_resource.severity)
  @current_resource.operinstruct(@new_resource.operinstruct)
  @current_resource.autoaction(@new_resource.autoaction)
  @current_resource.varbindsdecode(@new_resource.varbindsdecode)
  @current_resource.parameters(@new_resource.parameters)
  @current_resource.tticket(@new_resource.tticket)
  @current_resource.forward(@new_resource.forward)
  @current_resource.script(@new_resource.script)
  @current_resource.mouseovertext(@new_resource.mouseovertext)
  @current_resource.alarm_data(@new_resource.alarm_data)

  if ::File.exist?("#{node['opennms']['conf']['home']}/etc/#{@current_resource.file}")
    @current_resource.file_exists = true
    if event_file?(@current_resource.file, node)
      @current_resource.is_event_file = true
      if event_in_file?("#{node['opennms']['conf']['home']}/etc/#{@current_resource.file}", @current_resource)
        Chef::Log.debug("uei #{@current_resource.uei} is in file already")
        @current_resource.exists = true
        if event_changed?(@current_resource, node)
          Chef::Log.debug("uei #{@current_resource.uei} has changed.")
          @current_resource.changed = true
        end
      end
    end
  end
end
```
Note that it uses some library methods (`event_file?`, `event_in_file?`, `event_changed?`, etc), which you're free to do as well to keep the provider implementation file simpler.